src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,170 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.*;
+import io.opentracing.propagation.Format;
+
+import javax.imageio.spi.ServiceRegistry;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The <code>GlobalTracer</code> class is not a {@link Tracer} implementation of itself, but instead a utility-class
+ * to centrally reference the {@link #tracer() singleton instance} of the configured {@link Tracer} implementation.
+ * <p>
+ * The singleton instance can be instantiated in one of two ways:
+ * <ol>
+ * <li>Explicitly, by calling {@link #register(Tracer)} with a configured tracer implementation, or:</li>
+ * <li>Automatically by using the Java <code>ServiceLoader</code> SPI mechanism to load an implementation
+ * from the classpath.</li>
+ * </ol>
+ *
+ * @author Sjoerd Talsma
+ * @see Tracer
+ */
+public final class GlobalTracer {
```

Comment:
@adriancole For Scala I would not have attempted to write this library as the language itself has the concept of _implicits_, making it not nearly as painful to propagate contexts from call to call as it is in Java.
This is intended to relieve some of the explicit burden for the poor Java developers while still allowing any underlying Tracer implementation.